Classes/Plugin/Search.php

```diff
@@ -551,6 +551,26 @@ public function makeFacetsMenuArray($content, $conf) {
         $selectQuery = $solr->service->createSelect($search['params']);
         $results = $solr->service->select($selectQuery);
         $facet = $results->getFacetSet();
+
+        $facetCollectionArray = array();
+        foreach (\TYPO3\CMS\Core\Utility\GeneralUtility::trimExplode(',', $this->conf['facetCollections'], TRUE) as $facetCollection) {
```

Comment:
Instead of performing a single database query for every selected collection, I suggest doing one query instead (paraphrasing):
`SELECT index_name FROM tx_dlf_collections WHERE uid IN ($this-conf['facetCollections'])`

Since `$this->conf['facetCollections']` is a comma-separated list of UIDs, you can use it right away as a selector in the database query. Just do some sanitizing before to make sure that the variable indeed has the assumed format (i.e. contains only integers and commas).